src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala

```diff
@@ -95,50 +100,68 @@ trait UserService extends HttpService with PerRequestCreator with FireCloudReque
         }
       }
     } ~
     requireUserInfo() { userInfo =>
     pathPrefix("api") {
-      path("profile" / "billing") { requestContext =>
-        val extReq = Get(UserService.billingUrl)
-        externalHttpPerRequest(requestContext, extReq)
+      path("profile" / "billing") {
+        passthrough(UserService.billingUrl, HttpMethods.GET)
       } ~
-      path("profile" / "refreshTokenDate") { requestContext =>
-        val extReq = Get(OAuthService.remoteTokenDateUrl)
-        externalHttpPerRequest(requestContext, extReq)
+      path("profile" / "billingAccounts") {
+        oauthParams { (state, _) => requestContext =>
+          val pipeline = authHeaders(requestContext) ~> sendReceive
+          val extReq = Get(UserService.billingAccountsUrl)
+          pipeline(extReq) onComplete {
+            case Success(response) if response.status == Forbidden =>
+              val tryParseScopes = Try {
+                response.entity.asString.parseJson.convertTo[ErrorReport].message.parseJson.convertTo[BillingAccountScopes]
+              }
+              tryParseScopes match {
+                case Success(scopes) =>
+                  // user does not have appropriate scopes.  Ask user to enable them via OAuth
+                  val redirectURI = HttpGoogleServicesDAO.getGoogleRedirectURI(state, "auto", Option(scopes.requiredScopes ++ HttpGoogleServicesDAO.userLoginScopes))
+                  requestContext.complete(Forbidden, ErrorReport(Forbidden, BillingAccountRedirect(redirectURI).toJson.toString))
+                case _ =>
+                  requestContext.complete(Forbidden, response.entity)
+              }
+            case Success(response) => requestContext.complete(response.status, response.entity)
+            case Failure(regrets) => respondWithErrorReport(regrets, requestContext)
+          }
+        }
+      } ~
+      path("profile" / "refreshTokenDate") {
+        passthrough(OAuthService.remoteTokenDateUrl, HttpMethods.GET)
       }
     } ~
     pathPrefix("register") {
       pathEnd {
-        get { requestContext =>
-          externalHttpPerRequest(requestContext, Get(UserService.rawlsRegisterUserURL))
+        get {
+          passthrough(UserService.rawlsRegisterUserURL, HttpMethods.GET)
         }
       } ~
-      path("userinfo") { requestContext =>
-        val extReq = Get("https://www.googleapis.com/oauth2/v3/userinfo")
-        externalHttpPerRequest(requestContext, extReq)
+      path("userinfo") {
+        passthrough("https://www.googleapis.com/oauth2/v3/userinfo", HttpMethods.GET)
       } ~
       pathPrefix("profile") {
         // GET /profile - get all keys for current user
         pathEnd {
-          get { requestContext =>
-            val extReq = Get(UserService.remoteGetAllURL.format(userInfo.getUniqueId))
-            externalHttpPerRequest(requestContext, extReq)
+          get {
+            passthrough(UserService.remoteGetAllURL.format(userInfo.getUniqueId), HttpMethods.GET)
```

Comment:
I like the cleanup of using passthrough() instead of externalHttpPerRequest() ... but they're not exactly equal, iirc. passthrough() allows for subpaths - so this would open up /profile/{userid}/{key} as well as /profile/{userid}. I checked all the places you changed and I *think* they're fine.